### DISCUSSION TITLE
Strava upload?

### DISCUSSION THREAD

**Discussion by benjaminlin** [dvmarinoff/Auuki#116] [General]
Sorry I am new to the app, but I am really impressive how good it is since I was thinking to create one myself w/o paying Zwift for the summer time. Anyway, does anyone look into automatically Strava upload? Does Garmin upload will send the workout to Strava once uploaded? thanks!

**Comment by dvmarinoff** [comment]
@benjaminlin Hello and welcome to the repo!

I haven't looked at the Strava API, will set some time aside for this after I finish with the Workout Editor. Garmin Connect though  can upload to Strava automatically.

**Reply by benjaminlin** [reply]
@dvmarinoff  do you open to have others participating the project? if you do, I can work on this and figure out if it is possible to do, I played with Strava API long ago. thanks!

**Reply by dvmarinoff** [reply]
Yes, of course PRs are welcome.

**Reply by benjaminlin** [reply]
I am trying to work on Strava API response, what server side language I should use to parse the response? JS? php? or python? I am running local simple python server to test the API though, the activity upload will first asking user to authorize the upload, and the access token will send it back to server side, so I need to know what kind of server you are using so I know what language is supported. I maybe wrong on thinking about it because I have not done web development for years. thanks!

**Reply by dvmarinoff** [reply]
The project is hosted on [vercel](https://vercel.com/docs) and it's purely static. 'Vanilla' Javascript is the language of choice. Will be able to get back to you with more details for tomorrow.

**Reply by benjaminlin** [reply]
I just realized that yesterday when I was riding! ok...then we probably won't be able to do Strava auto upload until the project shift to dynamic web.

**Reply by dvmarinoff** [reply]
Kinda yes. But also it's time to start crossing that bridge. Let me finish with the Workout Editor feature, then I'll research some options, 'cause I really want to provide this sort of functionality free for the end User, but also sustainable for me in the long run.

**Reply by benjaminlin** [reply]
that will be awesome, let me know when you have something in mind for server side setup, we can go from there. In the meantime if you have anything you don't have time to do but would like someone to look into, please let me know.
